### ISSUE TITLE
🧩 [Phase 6] Training Loop + W&B Logging

### ISSUE THREAD

**Issue by shoaib-inamdar** [shoaib-inamdar/MultiModal-Cane-Disease-Detection#8]
**Parent:** #2 
## 🎯 Goal
Train the full model end-to-end with proper checkpointing, early stopping,
and live experiment tracking on Weights & Biases.
## 📋 Files to Create
- [ ] `src/training/trainer.py` — Main training loop
- [ ] `src/training/losses.py` — Cross-Entropy loss
- [ ] `src/training/schedulers.py` — CosineAnnealingLR
- [ ] `src/training/callbacks.py` — Early stopping + checkpoint save
- [ ] `scripts/train.py` — CLI entry point
## ✅ Done When
```bash
uv run python scripts/train.py --config configs/obj1_cross_attention.yaml
# Runs 3 epochs without error
# Loss decreases
# W&B dashboard shows loss + accuracy curves
